EIPS/eip-7928.md

```diff
@@ -0,0 +1,171 @@
+---
+eip: 7928
+title: Block-Level Access Lists
+description: Enforced block-level access lists for parallalizing transaction validation
+author: Toni Wahrstätter (@nerolation), Dankrad Feist (@dankrad), Francesco D`Amato (@fradamt), Jochem Brouwer (@jochem-brouwer), Ignacio Hagopian (@jsign), Yoav Weiss (@yoavw), Alex Forshtat (@forshtat)
+discussions-to: https://ethereum-magicians.org/t/eip-xxxx-block-level-access-lists/23337
+status: Draft
+type: Standards Track
+category: Core
+created: 2025-03-31
+---
+
+## Abstract
+
+This EIP proposes adding Block-Level Access Lists (BALs). By including a complete list of all addresses and storage keys accessed during a block, along with their post-execution values for writes, we enable parallel disk reads and parallel transaction validation. This improves execution efficiency and accelerates block verification, potentially allowing for future gas limit increases.
+
+## Motivation
+
+Currently, transactions without an explicit transaction access list cannot be efficiently parallelized, as the execution engine cannot determine in advance which addresses and storage slots will be accessed. While transaction-level access lists exist via EIP-2930, they are not enforced, making it difficult to optimize execution pipelines.
+
+By enforcing access lists at the block level, we shift the responsibility of declaring storage accesses to the block builder. This enables to efficiently parallelize both disk reads and transaction execution, knowing in advance the exact scope of storage interactions for each transaction.
+
```

Comment:
Might be useful to write a sentence with some TL;DR of your research on tx parallelization opportunity in mainnet? Given you did the research with real data, I think can give a less theoretical (i.e., stronger) argument on why parallelizing is possible. (e.g., in mainnet blocks between A and B, usually X% of txs are parallelizable, etc)

Reply:
agree, good point, will add.